src/components/Common/Environments/ModalCreate.vue

```diff
@@ -0,0 +1,204 @@
+          </div>
+        </div>
+        <div class="col s12">
+          <div class="row">
+            <div class="col s6 m3" v-for="color in colors">
+              <div class="color card valign-wrapper"
+                   :style="{backgroundColor: color}"
+                   @click="selectColor($index)">
+                <span class="selected valign center-align" v-if="environment.color === color">Selected</span>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <span slot="footer">
+        <button type="submit" class="waves-effect btn">
+            {{environmentId ? 'Update' : 'Create'}}
+        </button>
+        <button class="btn-flat waves-effect waves-grey" @click.prevent="$broadcast('modal-close', 'create-env')">
+            Cancel
+        </button>
+      </span>
+    </modal>
+  </form>
+</template>
+
+<style lang="scss" rel="stylesheet/scss" scoped>
+  .error {
+    strong {
+      display: block;
+    }
+  }
+  .input-field {
+    label {
+      width: 100%;
+      left: 0;
+    }
+  }
+  button {
+    &.btn-flat {
+      &:focus {
+        background-color: #EEE;
+      }
+    }
+  }
+  .last-row {
+    margin-bottom: 5px;
+  }
+  .color-picker {
+    p {
+      color: #9e9e9e;
+    }
+    .color {
+      display: block;
+      text-align: center;
+      cursor: pointer;
+      height: 60px;
+
+      &:hover {
+        box-shadow: 5px 10px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
+      }
+      .selected {
+        margin-top: 15px;
+        font-family: "Roboto", sans-serif;
+        letter-spacing: 1.5px;
+        text-transform: uppercase;
+        font-size: 1.2em;
+        font-weight: normal;
+        color: #fff;
+        display: inline-block;
+      }
+    }
+  }
+</style>
+
+<script>
+  import Modal from '../../Materialize/Modal'
+  import Focus from '../../../directives/focus.directive'
+  import { createEnvironment, updateEnvironment, DEFAULT_COLOR } from '../../../services/environment'
+  import { environments } from '../../../vuex/modules/common/kuzzle/getters'
+
+  export default {
+    name: 'EnvironmentsCreateModal',
+    props: ['environmentId'],
+    components: {
+      Modal
+    },
+    directives: {
+      Focus
+    },
+    vuex: {
+      getters: {
+        environments
+      }
+    },
+    data () {
+      return {
+        errors: {
+          wsPort: false,
+          ioPort: false,
+          host: false
+        },
+        environment: {
+          name: null,
+          host: null,
+          ioPort: 7512,
+          wsPort: 7513,
+          color: DEFAULT_COLOR
+        },
+        colors: [DEFAULT_COLOR, '#dc2222', '#7f6500', '#ccaa2a', '#537f00', '#000f7f', '#6d007f', '#7f0045']
+      }
+    },
+    methods: {
+      createEnvironments () {
+        this.errors.name = (!this.environment.name)
+        this.errors.wsPort = (!this.environment.wsPort || this.environment.wsPort < 1000 || this.environment.wsPort > 9999)
```

Comment:
This is a form validation. I think the code must be in the component where the form is